.github/workflows/main.yml

```diff
@@ -12,7 +12,17 @@ on:
       - main
 
 jobs:
-  build:
+  check-package:
+    name: Build & inspect our package.
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```

Comment:
Out of curiosity, why is this needed?

And if is needed, why we don't use fetch-depth=0 in the `test` job?

Reply:
this is needed to get the version of the package correctly for setuptools_scm - build doesn't need it as its using artifacts